core/controllers/reader.py

```diff
@@ -330,12 +330,19 @@ def post(self, unused_exploration_id):
         """
         # A domain object representing the old state.
         old_state = exp_domain.State.from_dict(self.payload.get('old_state'))
+        # The id of the exploration to which the state belongs.
+        exp_id = self.payload.get('exp_id')
+        # The version of the exploration to which the state belongs.
+        exp_version = self.payload.get('exp_version')
+        # The name of the State to which the interaction belongs.
+        state_name = self.payload.get('state_name')
         # The learner's raw answer.
         answer = self.payload.get('answer')
         # The learner's parameter values.
         params = self.payload.get('params')
         params['answer'] = answer
-        result = classifier_services.classify(old_state, answer)
+        result = classifier_services.classify(old_state, answer, exp_id,
+                                              exp_version, state_name)
```

Comment:
just for maintaining consistency, I think it would be better to rename exp_version to exp_version_when_created (that's what we use in storage layer).